pkg/net/libp2p/channels_fuzzing_test.go

```diff
@@ -0,0 +1,89 @@
+package libp2p
+
+import (
+	"context"
+	"testing"
+
+	fuzz "github.com/google/gofuzz"
+	"github.com/keep-network/keep-core/pkg/net"
+)
+
+func TestUnicastChannelFuzzing(t *testing.T) {
+	ctx := context.Background()
+
+	withNetwork(ctx, t, 7000, func(
+		_ *identity,
+		identity2 *identity,
+		provider1 net.Provider,
+		provider2 net.Provider,
+	) {
+		channel, err := provider1.UnicastChannelWith(identity2.id)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		provider2.OnUnicastChannelOpened(func(channel net.UnicastChannel) {
+			channel.SetUnmarshaler(func() net.TaggedUnmarshaler {
+				return &fuzzingMessage{}
```

Comment:
Per our [previous discussion](https://github.com/keep-network/keep-core/pull/1599/files#r408862061), I think we should rather fuzz on `*pb.UnicastNetworkMessage` as this type is what we are processing in `processMessage` of `unicastChannel`. Random bytes are fine but I don't think it give us any advantage to generate more than 10 random cases. 

I think we should also call `processMessage` directly. We are probably more interested in fuzz-testing receive part than the send part. All marshallers will have a roundtrip fuzz tests and send is covered with unit tests for a happy path, I think it's enough.

Reply:
Same for the broadcast channel part below.

Reply:
We are not only testing the send part here but the receive part as well because each generated byte slice pass through the whole channel. Though, I think we can direct `processMessage` fuzzing for better coverage.

Reply:
OK, let's leave it as-is, this is covering `handleStream` function in the unicast channel and `processPubsubMessage` in the broadcast channel. Though I think we should reduce it to the minimum, e.g. `10` fuzzing iterations with `0.1` nil chance and add separate fuzzing for `processMessage` as you suggested.